std/array.d

```diff
@@ -380,6 +380,113 @@ if (isNarrowString!String)
     )));
 }
 
+/**
+ * Returns a static array from the stack with length of `s` and initializes
+ * it with copies of the elements of range $(D r). The remainder of the
+ * array is default-initialized.
+ *
+ * Recommended particulary for complex initializations of static arrays
+ * during compilation.
+ *
+ * Params:
+ *      r = The range (or aggregate with $(D opApply) function) whose elements
+ *      are copied into the stack-allocated array. Must be as long or shorter
+ *      than `s`, otherwise causes an array overflow.
```

Comment:
Please no overflow. It's trivial and low cost to have a test here. In case of overflow, just fill the array and place the real length in `rangeLength`.

Reply:
Take would be easy to add if the user wants it, but not to remove, so this makes it more-than-zero cost abstraction.

On the other hand, we can always add a non-checking equivalent later so I tend to agree with this. Regardless, I'll do it.